substrate/frame/revive/src/exec.rs

```diff
@@ -1836,24 +1844,35 @@ where
 		// This is for example the case when creating the frame fails.
 		*self.last_frame_output_mut() = Default::default();
 
-		let executable = E::from_storage(code_hash, self.gas_meter_mut())?;
-		let sender = &self.top_frame().account_id;
-		let executable = self.push_frame(
-			FrameArgs::Instantiate {
-				sender: sender.clone(),
-				executable,
-				salt,
-				input_data: input_data.as_ref(),
-			},
-			value,
-			gas_limit,
-			deposit_limit.saturated_into::<BalanceOf<T>>(),
-			self.is_read_only(),
-		)?;
+		let executable = {
+			let executable = E::from_storage(code_hash, self.gas_meter_mut())?;
+			let sender = &self.top_frame().account_id;
+			self.push_frame(
+				FrameArgs::Instantiate {
+					sender: sender.clone(),
+					executable,
+					salt,
+					input_data: input_data.as_ref(),
+				},
+				value,
+				gas_limit,
+				deposit_limit.saturated_into::<BalanceOf<T>>(),
+				self.is_read_only(),
+			)?
+		};
+		let executable = executable.expect(FRAME_ALWAYS_EXISTS_ON_INSTANTIATE);
+		// Mark the contract as created in this transaction
+		// Get the trie_id from the newly created frame's contract info
+		if let ExecutableOrPrecompile::<T, E, Self>::Executable(_) = &executable {
+			let trie_id = {
+				let contract_info = self.top_frame_mut().contract_info();
+				contract_info.trie_id.clone()
+			};
+			self.contracts_created.insert(trie_id);
+		}
```

Comment:
why not using the address for the contracts_created?

I would just keep the existing code, push the address before in the hash
and then remove it if in map_err

Reply:
you are only going to strore in this set the constructed contract, which can be referenced by the address.
if the call fails here you should remove the freshly entered address in the set, map_err or maybe inspect_err, is a good place to do that